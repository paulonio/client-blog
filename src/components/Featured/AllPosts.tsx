import React, { FC } from 'react';

import Link from 'next/link';
import { PostType } from '@/types/types';
import { sen } from '@/styles/fonts';

import styles from './styled.module.scss';

interface AllPostsProps {
  posts: PostType[];
}

const AllPosts: FC<AllPostsProps> = ({ posts }) => {
  return (
    <div className={styles.posts}>
      <div className={styles.posts__header}>
        <h2 className={sen.className}>All Posts</h2>
        <Link href="/blog">
          <button type="button" className={`body-s ${styles.link__button}`}>
            View All
          </button>
        </Link>
      </div>
      <div className={styles.posts__wrapper}>
        {posts.map((post) => (
          <div className={styles.post__container}>
            <p className="body-l">
              By <span className={styles.name}>{post.username}</span> | {post.date}
            </p>
            <h4 className={sen.className}>{post.title}</h4>
          </div>
        ))}
      </div>
    </div>
  );
};

export default AllPosts;
