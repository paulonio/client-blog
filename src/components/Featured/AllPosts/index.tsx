import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';
import Link from 'next/link';

import { PostType } from '@/types/types';
import { sen } from '@/styles/fonts';

import styles from './styled.module.scss';
import Button from '../../Button';
import { stringToKey } from '@/utils/utils';

interface AllPostsProps {
  posts: PostType[];
}

const AllPosts: FC<AllPostsProps> = ({ posts }) => {
  const { t } = useTranslation(['featured', 'common']);

  return (
    <div className={styles.posts}>
      <div className={styles.posts__header}>
        <h2 className={sen.className}>{t('all')}</h2>
        <Link href="/blog">
          <Button buttonType="link">{t('view')}</Button>
        </Link>
      </div>
      <div className={styles.posts__wrapper}>
        {posts.map((post) => (
          <div key={post.id} className={styles.post__container}>
            <p className="body-l">
              {t('by', { ns: 'common' })}{' '}
              <span className={styles.name}>{t(stringToKey(post.username), { ns: 'common' })}</span>{' '}
              | {t(stringToKey(post.date), { ns: 'common' })}
            </p>
            <h4 className={sen.className}>{t(stringToKey(post.title))}</h4>
          </div>
        ))}
      </div>
    </div>
  );
};

export default AllPosts;
