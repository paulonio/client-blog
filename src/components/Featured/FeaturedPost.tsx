import React, { FC } from 'react';

import Button from '@/components/Button';

import type { Post } from '@/types/types';
import { sen } from '@/styles/fonts';

import styles from './styled.module.scss';

interface SinglePostProps {
  post: Post;
}

const FeaturedPost: FC<SinglePostProps> = ({ post }) => {
  const { username, date, title, body, urlToImage } = post;

  return (
    <div className={styles.featured__post}>
      <h2 className={sen.className}>Featured Post</h2>
      <div className={styles.post__wrapper}>
        <div className={styles.image}>
          <img src={urlToImage} alt="Featured" />
        </div>
        <div className={styles.post__content}>
          <p className="body-l">
            By <span className={styles.name}>{username}</span> | {date}
          </p>
          <h3 className={sen.className}>{title}</h3>
          <p className="body-s">{body}</p>
        </div>
        <Button>Read More {'>'}</Button>
      </div>
    </div>
  );
};

export default FeaturedPost;
