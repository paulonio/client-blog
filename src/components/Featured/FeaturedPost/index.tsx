import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';
import Link from 'next/link';

import Button from '@/components/Button';

import { stringToKey } from '@/utils/utils';
import type { PostType } from '@/types/types';
import { sen } from '@/styles/fonts';

import styles from './styled.module.scss';

interface SinglePostProps {
  post: PostType;
}

const FeaturedPost: FC<SinglePostProps> = ({ post }) => {
  const { id, username, date, title, body, urlToImage } = post;
  const { t } = useTranslation(['featured', 'common']);

  return (
    <div className={styles.featured__post}>
      <h2 className={sen.className}>{t('title')}</h2>
      <div className={styles.post__wrapper}>
        <div className={styles.image}>
          <img src={urlToImage} alt="Featured" />
        </div>
        <div className={styles.post__content}>
          <p className="body-l">
            {t('by', { ns: 'common' })}{' '}
            <span className={styles.name}>{t(stringToKey(username), { ns: 'common' })}</span> |{' '}
            {t(stringToKey(date), { ns: 'common' })}
          </p>
          <h3 className={sen.className}>{t(stringToKey(title))}</h3>
          <p className="body-s">{t(stringToKey(body))}</p>
        </div>
        <Link href={`/blog/${id}`}>
          <Button>
            {t('read')} {'>'}
          </Button>
        </Link>
      </div>
    </div>
  );
};

export default FeaturedPost;
