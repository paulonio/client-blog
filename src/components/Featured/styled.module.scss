@import '../../styles/sass/constants';

.featured {
  margin-top: 128px;
  display: flex;
  justify-content: space-between;

  &__post {
    border: solid 1px rgba($color-dark-s, 0.1);
  }

  @media screen and (max-width: 900px) {
    flex-direction: column;
    align-items: center;
    row-gap: 16px;
  }
}

.post {
  &__wrapper {
    max-width: 733px;
    width: 100%;
    margin-top: 32px;
    padding: 32px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    row-gap: 32px;
  }

  &__content {
    display: flex;
    flex-direction: column;
    row-gap: 16px;
  }

  &__container {
    padding: 32px 32px 25px;
    display: flex;
    flex-direction: column;
    align-items: start;
    row-gap: 8px;
    transition: background-color 0.3s;

    &:hover {
      background-color: $color-light-s;
      cursor: pointer;
    }
  }
}

.posts {
  max-width: 517px;
  width: 100%;
  
  &__header {
    display: flex;
    align-items: center;
    justify-content: space-between;

    @media screen and (max-width: 1280px) {
      padding: 0 6px;
    }
  }

  &__wrapper {
    display: flex;
    flex-direction: column;
  }
}

.image {
  & img {
    width: 100%;
  }
}

.name {
  color: $color-secondary;
}