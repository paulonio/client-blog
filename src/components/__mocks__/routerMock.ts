import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

(useRouter as jest.Mock).mockReturnValue({ query: {} });
