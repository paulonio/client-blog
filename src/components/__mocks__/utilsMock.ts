import { isLinkActive, parseString } from '@/utils/utils';

jest.mock('../../utils/utils', () => ({
  isLinkActive: jest.fn(),
  parseString: jest.fn(),
}));

(isLinkActive as jest.Mock).mockReturnValue(true);
(parseString as jest.Mock).mockReturnValue('');
