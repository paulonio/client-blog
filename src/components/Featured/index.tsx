import React from 'react';

import FeaturedPost from './FeaturedPost';
import AllPosts from './AllPosts';

import { POSTS } from '@/constants/posts';

import styles from './styled.module.scss';

const Featured = () => {
  const [post] = POSTS;
  const posts = POSTS.slice(0, 4);

  return (
    <section className={styles.featured}>
      <FeaturedPost post={post} />
      <AllPosts posts={posts} />
    </section>
  );
};

export default Featured;
