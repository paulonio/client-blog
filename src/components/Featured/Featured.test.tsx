import React from 'react';
import { render } from '@testing-library/react';
import Featured from '.';
import AllPosts from './AllPosts';
import { PostType } from '@/types/types';
import '../__mocks__/intersectionObserverMock';

const TEST_POSTS: PostType[] = [
  {
    id: 'test',
    body: 'test',
    category: 'test',
    date: 'test',
    tags: ['test'],
    title: 'test',
    urlToAvatar: 'test',
    urlToImage: 'test',
    username: 'test',
  },
];

describe('Featured', () => {
  it('should render featured component', () => {
    const { container } = render(<Featured />);

    expect(container).toMatchSnapshot();
  });

  it('should render all posts component', () => {
    const { container } = render(<AllPosts posts={TEST_POSTS} />);

    expect(container).toMatchSnapshot();
  });
});
