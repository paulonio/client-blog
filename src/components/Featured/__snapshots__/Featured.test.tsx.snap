// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Featured should render all posts component 1`] = `
<div>
  <div
    class="posts"
  >
    <div
      class="posts__header"
    >
      <h2
        class="className"
      >
        all
      </h2>
      <a
        href="/blog"
      >
        <button
          class="body-s link__button"
          type="button"
        >
          view
        </button>
      </a>
    </div>
    <div
      class="posts__wrapper"
    >
      <div
        class="post__container"
      >
        <p
          class="body-l"
        >
          by
           
          <span
            class="name"
          >
            test
          </span>
           
          | 
          test
        </p>
        <h4
          class="className"
        >
          test
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`Featured should render featured component 1`] = `
<div>
  <section
    class="featured"
  />
</div>
`;
